### PR TITLE
Fix spell check

### PR DESCRIPTION
**Pull request by hendursaga**
The mode is still not where I'd like it to be, but at least it works again as it did before!

### CODE REVIEW

File: source/mode/spell-check.lisp [hunk 1]
```diff
@@ -24,51 +24,55 @@ a list of more languages available, see `enchant:broker-list-dicts'.")
       (list
        "z =" 'spell-check-word)))))
 
+(defmacro with-spell-check ((variable) &body body)
+  `(enchant:with-dict (,variable (spell-check-language
+                                  (find-submode 'spell-check-mode)))
+     ,@body))
+
+(defun spell-dict-check (word)
```
Comment (Ambrevar) [inline source/mode/spell-check.lisp:32]: It's a predicate then.  Suffix it with `-p` to make this clear.

Optional: Declaim the function type, e.g. `(-> spell-dict-check (string) boolean)`

**Comment by Ambrevar** [comment]
This is excellent, thank you so much!

Could you address the nit I've reported, then I'll merge!

**Comment by hendursaga** [comment]
I've decided to not declaim, under SBCL I ran into...

```
; note: Type assertion too complex to check:
; (VALUES &OPTIONAL BOOLEAN &REST T).
; It allows an unknown number of values, consider using
; (VALUES &OPTIONAL BOOLEAN).
```

... and after a few attempts to fix that, I decided it wasn't worth it. I learned more about the type system though! :)

You can merge.

**Comment by Ambrevar** [comment]
Thanks a lot!

**Pull request merged**